### PR TITLE
BE-73 Change ids from int to uuid together with endpoints with migrations

### PR DESCRIPTION
**Pull request by KToldy**
Done with 4 migrations, changes endpoints and database from int ids to uuid.

### CODE REVIEW

File: antiplag/tests/test_views.py [hunk 1]
```diff
@@ -31,24 +31,28 @@ def test_handle_duplicates(self):
             document=self.document2,
             matched_docs=[
                 {"name": "test_name", "elastic_id": 123, "percentage": 0.19},
-                {"name": "test_name", "id": result1.id, "percentage": 0.19},
+                {"name": "test_name", "id": str(self.document.id), "percentage": 0.19},
```
Comment (MCFreddie777) [inline antiplag/tests/test_views.py:34]: I don't think this is correct from functional and logic perspective. Previously, the matched docs contained the id from result1, and now you changed it into document?

File: antiplag/migrations/0014_migrate_uuid_foreign_keys.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                related_name="documents",
+                to="antiplag.Submission",
+            ),
+        ),
+        migrations.AddField(
+            model_name="Result",
+            name="document",
+            field=models.OneToOneField(
+                on_delete=models.CASCADE,
+                null=True,
```
Comment (OliverKanat) [inline antiplag/migrations/0014_migrate_uuid_foreign_keys.py:39]: Looks like you forgot final migration to remove `null=True`.

**Review by OliverKanat — CHANGES_REQUESTED** [review]

File: antiplag/tests/test_views.py (hunk 1 above)
Comment (OliverKanat) [reply antiplag/tests/test_views.py:34]: Looks like previous tests were not correct. `matched_docs` should have IDs of documents after all, not IDs of results.

File: antiplag/migrations/0014_migrate_uuid_foreign_keys.py (hunk 2 above)
Comment (KToldy) [reply antiplag/migrations/0014_migrate_uuid_foreign_keys.py:39]: You're right, added one last migration.

File: antiplag/tests/test_views.py (hunk 1 above)
Comment (MCFreddie777) [reply antiplag/tests/test_views.py:34]: Got it. Good job for finding it.

**Review by MCFreddie777 — APPROVED** [review]
Not sure about those migrations, but if @OliverKanat approves it, looks good to me.

File: antiplag/migrations/0015_remove_null.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            field=models.JSONField(encoder=antiplag.encoders.UUIDEncoder),
+        ),
+        migrations.AlterField(
+            model_name="submission",
+            name="id",
+            field=models.UUIDField(
+                default=uuid.uuid4, editable=False, primary_key=True, serialize=False
+            ),
+        ),
+    ]
```
Comment (OliverKanat) [inline antiplag/migrations/0015_remove_null.py:51]: Why are you altering all these fields?

Comment (KToldy) [reply antiplag/migrations/0015_remove_null.py]: Done, was unique changed.

**Review by MCFreddie777 — APPROVED** [review]

**Review by OliverKanat — APPROVED** [review]

**Pull request merged**